src/benchmarklib/CMakeLists.txt

```diff
@@ -18,18 +15,35 @@ set(
     tpch/tpch_queries.hpp
     tpch/tpch_query_generator.cpp
     tpch/tpch_query_generator.hpp
-    tpch/tpch_db_generator.cpp
-    tpch/tpch_db_generator.hpp
-
-    table_generator.cpp
-    table_generator.hpp
+    tpch/tpch_table_generator.cpp
+    tpch/tpch_table_generator.hpp
 
-    abstract_query_generator.cpp
+    abstract_benchmark_table_generator.hpp
+    abstract_table_generator.cpp
+    abstract_table_generator.hpp
     abstract_query_generator.hpp
+    abstract_query_generator.cpp
+    benchmark_config.cpp
+    benchmark_config.hpp
     benchmark_runner.cpp
     benchmark_runner.hpp
-    benchmark_utils.cpp
-    benchmark_utils.hpp
+    benchmark_state.cpp
+    benchmark_state.hpp
+    benchmark_table_encoder.cpp
+    benchmark_table_encoder.hpp
+    cli_config_parser.cpp
+    cli_config_parser.hpp
+    encoding_config.cpp
+    encoding_config.hpp
+    file_based_table_generator.cpp
+    file_based_table_generator.hpp
+    file_based_query_generator.cpp
```

Comment:
Why is this in the benchmarklib project?

Reply:
I thought about directly linking it to the filebasedbenchmark as well. But then again you couldn't use this from the playground and I dislike not being able to fiddle around without everything I want and having to alter buildsystems before I can.